formats/properties/commonMain/src/kotlinx/serialization/properties/StringProperties.kt

```diff
@@ -0,0 +1,291 @@
+ * // data contents will be the following:
+ * // DataHolder(data = Data(property1 = "value1"), property2 = "value2")
+ * ```
+ *
+ * @param conf A [PropertiesConf] which contain configuration for customising the output string.
+ */
+@ExperimentalSerializationApi
+@Suppress("UNUSED_PARAMETER")
+public sealed class StringProperties(
+    private val conf: PropertiesConf,
+    private val properties: Properties = Properties(conf.serializersModule),
+) : SerialFormat by properties, StringFormat {
+
+    /**
+     * Encodes properties from the given [value] to a properties String using the given [serializer].
+     * `null` values are omitted from the output.
+     */
+    @ExperimentalSerializationApi
+    public override fun <T> encodeToString(serializer: SerializationStrategy<T>, value: T): String {
+        val map = properties.encodeToMap(serializer, value)
+        val builder = StringBuilder()
+        for ((k, v) in map) {
+            builder.append(k)
+            repeat(conf.spacesBeforeSeparator) {
+                builder.append(' ')
+            }
+            builder.append(conf.keyValueSeparator.char())
+            repeat(conf.spacesAfterSeparator) {
+                builder.append(' ')
+            }
+            builder.append(v)
+            builder.append(conf.lineSeparator.chars())
+        }
+        return builder.toString()
+    }
+
+    /**
+     * Decodes properties from the given [string] to a value of type [T] using the given [deserializer].
+     * [String] values are converted to respective primitive types using default conversion methods.
+     * [T] may contain properties of nullable types; they will be filled by non-null values from the [map], if present.
+     */
+    public override fun <T> decodeFromString(deserializer: DeserializationStrategy<T>, string: String): T {
+        val result = mutableMapOf<String, String>()
+        for (line in string.logicalLines()) {
+            val parsedLine = line.unescaped()
+            var keyEnd = parsedLine.length
+            for (i in parsedLine.indices) {
+                if (parsedLine[i] in separators) {
+                    keyEnd = i
+                    break
+                }
+            }
+
+            var valueBegin = parsedLine.length
+            var separatorFound = false
+            for (i in keyEnd..parsedLine.lastIndex) {
+                if (separatorFound && parsedLine[i] != ' ') {
+                    valueBegin = i
+                    break
+                }
+                if (parsedLine[i] in nonBlankSeparators) {
+                    separatorFound = true
+                }
+                if (parsedLine[i] !in separators) {
+                    valueBegin = i
+                    break
+                }
+            }
+
+            result[parsedLine.substring(0, keyEnd)] = parsedLine.substring(valueBegin)
+        }
+        return properties.decodeFromStringMap(deserializer, result)
```

Comment:
Same here, I would add an overload to get the input string as `Map<String, String` too.